### PR TITLE
fix: Restrict event save when information is insufficient

### LINKED ISSUE #2920: Only allow "Publish Event" If the Event has Relevant Info
A user should always be able to save a draft with basic info. However in order to publish an event the following info should be required:
* Name of Event
* Location
* Dates
* At least one Ticket exists

Show an error message if the event does not have sufficient data to get published.

Please delete the option to show events from other platforms using a URL for now

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #2920

#### Short description of what this resolves:
The event save action will send appropriate error message if the following cases fail in the client level
- Name of Event
- Location
- Dates
- At least one Ticket exists

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by uds5501** [comment]
@kushthedude 
I thought a frontend application was required as well to handle this issue as per https://github.com/fossasia/open-event-frontend/issues/2920#issuecomment-501560136

**Comment by kushthedude** [comment]
> @kushthedude
> I thought a frontend application was required as well to handle this issue as per [#2920 (comment)](https://github.com/fossasia/open-event-frontend/issues/2920#issuecomment-501560136)

Arent the Semantic UI Validators enough for such ?

File: app/mixins/event-wizard.js [hunk 1]
```diff
@@ -59,59 +59,76 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
+      await Promise.all(bulkPromises);
 
-    return event;
+      return event;
+    } else {
+      let errorObject = { 'errors': [] };
+      if (event.name === undefined || event.name === '') {
+        errorObject.errors.push({ 'detail': 'Event name has not been provided' });
+      }
+      if (event.locationName === undefined || event.locationName === '') {
```
Comment (kushthedude) [inline app/mixins/event-wizard.js]: These are already covered by Form Validators, arent they ?

File: app/mixins/event-wizard.js [hunk 2]
```diff
@@ -128,8 +145,9 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
         this.transitionToRoute(route, data.id);
       })
       .catch(e => {
-        console.error(e);
-        this.notify.error(this.l10n.t(e.errors[0].detail));
+        e.errors.forEach(error => {
+          this.notify.error(this.l10n.t(error.detail));
```
Comment (CosmicCoder96) [inline app/mixins/event-wizard.js]: use `tVar` for dynamic translations.

File: app/mixins/event-wizard.js [hunk 3]
```diff
@@ -59,59 +59,76 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
...
+      return event;
+    } else {
+      let errorObject = { 'errors': [] };
+      if (event.name === undefined || event.name === '') {
+        errorObject.errors.push({ 'detail': 'Event name has not been provided' });
+      }
+      if (event.locationName === undefined || event.locationName === '') {
+        errorObject.errors.push({ 'detail': 'Location has not been provided' });
+      }
+      if (event.startsAtDate === undefined || event.endsAtDate === undefined) {
```
Comment (mrsaicharan1) [inline app/mixins/event-wizard.js]: Can we have a check for ends >= starts ?

Comment (uds5501) [reply app/mixins/event-wizard.js]: @mrsaicharan1 That's a schema check and already handled on server, besides I am trying to keep this PR aligned with the FE issue

File: app/mixins/event-wizard.js [hunk 4]
```diff
@@ -59,59 +59,76 @@ export default Mixin.create(MutableArray, CustomFormMixin, {
         }
       }
     }
+    const numberOfTickets = data.tickets.length;
```
Comment (CosmicCoder96) [inline app/mixins/event-wizard.js]: This code will completely crash if for some reason data.tickets is not defined.
Every where else, first it is checked if data.property exists. Data.undefined.length will throw an exception.

Comment (uds5501) [reply app/mixins/event-wizard.js]: @CosmicCoder96 Understood, changing accordingly

Comment (uds5501) [reply app/mixins/event-wizard.js]: @CosmicCoder96 Updated the assignment, please check now

**Comment by uds5501** [comment]
@mrsaicharan1 @shreyanshdwivedi @CosmicCoder96 @kushthedude 
I have finalised the changes, please review

**Comment by uds5501** [comment]
@mrsaicharan1 @shreyanshdwivedi please take a look

**Comment by uds5501** [comment]
@CosmicCoder96 Rebased the commits.

**Pull request merged**